src/main/java/com/internet/shop/dao/OrderDao.java

```diff
@@ -0,0 +1,22 @@
+package com.internet.shop.dao;
+
+import com.internet.shop.model.Order;
+import com.internet.shop.model.ShoppingCart;
+import java.util.List;
+import java.util.Optional;
+
+public interface OrderDao {
+    Order completeOrder(ShoppingCart shoppingCart);
```

Comment:
we should have this method only on the service layer
there we should have only create() method

Reply:
Yes, messed up packages. Done